src/mapgen_functions.cpp

```diff
@@ -867,6 +867,17 @@ void mapgen_road( mapgendata &dat )
         }
     }
 
+    // place street lights
+    if( neighbor_sidewalks ) {
+        for( int x = 0; x < SEEX * 2 - 1; x++ ) {
+            for( int y = 0; y < SEEY * 2 - 1; y++ ) {
+                if( x % 5 == 0 && y % 5 == 0 && m->ter( point( x, y ) ) == t_sidewalk ) {
```

Comment:
Why such a long IF statement?

Reply:
Because I'm not good at algorithms at all.